doc/source/serve/app-builder-guide.md

```diff
@@ -0,0 +1,140 @@
+# Passing Arguments to Applications
+
+This section describes how to pass arguments to your applications using an application builder function.
```

Comment:
One concern I have is that we never introduce/formally define the concept of "applications" to users (in key concepts)

Reply:
you're totally right, I have it on my TODO list to fix that (the getting started guide is updated to reflect this verbiage already).

let me see if I can get that change in for 2.5 as well, should be straightforward.

Reply:
Could we add an entry to the Glossary, too? https://docs.ray.io/en/latest/ray-references/glossary.html